runtime/memprof.c

```diff
@@ -87,95 +87,42 @@ static double mt_generate_uniform(void)
           1.16415321826934814453125e-10; /* 2^-33 */
 }
 
-/* C99's [lgammaf] function is not available in some compilers
-   (including MSVC). Here is our own approximate implementation of the
-   log-factorial function.
-
-   Requirement: [n] is a non-negative integer.
-
-   We use Ramanujan's formula. For n < 10^8, the absolute error is
-   less than 10^-6, which is way better than what we need.
- */
-static double lfact(double n)
+/* Simulate a geometric variable of parameter [lambda].
+   The result is clipped in [0..Max_long]
```

Comment:
[1..Max_long]

(this is the version of the geometric distribution that measures "number of attempts to first success" rather than "number of failures before first success")